zipline/lib/adjusted_array.py

```diff
@@ -82,19 +90,29 @@ def _normalize_array(data):
 
     Returns
     -------
-    coerced, viewtype : (np.ndarray, np.dtype)
+    coerced, view_kwargs : (np.ndarray, np.dtype)
     """
+    if isinstance(data, LabelArray):
+        return data, {}
+
     data_dtype = data.dtype
     if data_dtype == bool_:
-        return data.astype(uint8), dtype(bool_)
+        return data.astype(uint8), {'dtype': dtype(bool_)}
     elif data_dtype in FLOAT_DTYPES:
-        return data.astype(float64), dtype(float64)
+        return data.astype(float64), {'dtype': dtype(float64)}
     elif data_dtype in INT_DTYPES:
-        return data.astype(int64), dtype(int64)
-    elif data_dtype.name.startswith('datetime'):
+        return data.astype(int64), {'dtype': dtype(int64)}
+    elif data_dtype in CATEGORICAL_DTYPES:
```

Comment:
Why doesn't `CATEGORICAL_DTYPES` include 'S' and 'U'? I was under the impression that `LabelArray` can accept data of strings/unicode, not just dtype `object`.